akka-http-core/src/main/scala/akka/http/scaladsl/ServerBuilder.scala

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright (C) 2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.http.scaladsl
+
+import akka.actor.ClassicActorSystemProvider
+import akka.event.LoggingAdapter
+import akka.http.scaladsl
+import akka.http.scaladsl.Http.ServerBinding
+import akka.http.scaladsl.model.{ HttpRequest, HttpResponse }
+import akka.http.scaladsl.settings.ServerSettings
+import akka.stream.scaladsl.{ Flow, Source }
+
+import scala.concurrent.Future
+
+/**
+ * The new API to create server bindings
+ */
+trait ServerBuilder {
+  def onPort(port: Int): ServerBuilder
+  def logTo(log: LoggingAdapter): ServerBuilder
+  def settings(settings: ServerSettings): ServerBuilder
+  def adaptSettings(f: ServerSettings => ServerSettings): ServerBuilder
+  def enableHttps(context: HttpsConnectionContext): ServerBuilder
+
+  def bind(): Source[Http.IncomingConnection, Future[ServerBinding]]
+  def bindFlow(handlerFlow: Flow[HttpRequest, HttpResponse, _]): Future[ServerBinding]
```

Comment:
This cannot be an overload called `bind` because otherwise, it's ambiguous for routes for conversions are defined both for flow and function (not a blocker for me, we want to deemphasize the flow variant anyway).